src/main/java/com/iota/iri/cache/Cache.java

```diff
@@ -0,0 +1,117 @@
+package com.iota.iri.cache;
+
+import com.iota.iri.cache.util.ValueWrapper;
+import com.iota.iri.storage.PersistenceProvider;
+
+import java.util.Collection;
+import java.util.Map;
+
+public interface Cache {
+
+    /**
+     * Get the name of this cache.
+     * 
+     * @return The name of the cache
+     */
+    String getName();
+
+    /**
+     * Get the value mapped to the specified key. Returns {@code null} if the mapping is not found.
+     * 
+     * @param key The keys whos value is to be returned
+     * @return The mapped value of the specified key
+     */
+    ValueWrapper get(Object key);
+
+    /**
+     * Get the value mapped to the specified key cast to the requested type. This does not differentiate between a
+     * cached {@code null} and no cached value at all. If the value is {@code null}, the return type is therefore
+     * irrelevant. If the value does not match the requested type, an {@code IllegalStateException} is thrown.
+     * 
+     * @param key  The key whos value is to be returned
+     * @param type The type of the value to match
+     * @param <T>  The class type
+     * @return The mapped value of the specified key.
+     */
+    <T> T get(Object key, Class<T> type) throws IllegalStateException;
+
+    /**
+     * Get the value mapped to the specified key obtaining the value from the persistence provider if not found in
+     * cache. If cache return; Otherwise db read, cache and return.
+     * 
+     * @param key                 The key wohs value is to be returned
+     * @param persistenceProvider
+     * @param <T>                 The class type of the returned value
+     * @return The mapped value of the specified key
+     */
+    <T> T get(Object key, PersistenceProvider persistenceProvider);
```

Comment:
I imagined that currently the `Tangle` class will hold the caches and persistence providers and handle the data transfer between them.

I don't think the `Cache` should access the Persistence Provider directly

I don't think the cache should know about the Persistence Provider. Remember that there may be data that we decide not too cache as well.